### ISSUE TITLE
3rd parameter of beforeOnCellMouseDown event can be TH, not only TD

### ISSUE THREAD

**Issue by AMBudnik** [handsontable/docs#109]
### Description
3rd parameter of beforeOnCellMouseDown event can be TH, not only TD.
It happens when we click a header of a column or row.

However, the documentation states that we will get a TD element.

![Zrzut ekranu 2020-05-5 o 10 56 05](https://user-images.githubusercontent.com/10757813/81049739-13c81980-8ebf-11ea-82f4-0e69b981ebd4.png)

### Demo
https://jsfiddle.net/trhznf17/ (Handsontable v7.4.2)

### Source
https://handsontable.com/docs/7.4.2/Hooks.html#event:beforeOnCellMouseDown

**Comment by scarletfog** [comment]
@wojciechczerniak this is a bit quirky to me, indeed we have "Cell's TD (or TH)" but it is a description of a... `TD` param?

**Comment by wojciechczerniak** [comment]
Param name can be anything you want. TD is probably for simplicity and compatibility with other APIs. The type is correct, both HTMLTable**Data**CellElement and HTMLTable**Header**CellElement inherit from HTMLTableCellElement (see the spec and note below IDL: https://html.spec.whatwg.org/multipage/tables.html#htmltablecellelement). You may improve the description

**Comment by scarletfog** [comment]
"TD is probably for simplicity and compatibility with other APIs" got it! 👍 @wojciechczerniak

**Comment by AMBudnik** [comment]
Should we then remove info about TH as a possible result? Example `afterOnCellContextMenu`- https://handsontable.com/docs/8.0.0/Hooks.html#event:beforeOnCellMouseDown

![Zrzut ekranu 2020-09-17 o 08 45 20](https://user-images.githubusercontent.com/10757813/93429908-2701e180-f8c2-11ea-9828-dc9119268fea.png)

**Comment by wojciechczerniak** [comment]
@AMBudnik No, the description is ok. Type HTMLTableCellElement is for both TD and TH.
